integrations/drip/drip.go

```diff
@@ -1,31 +1,129 @@
+}
+
 type Drip struct {
+	api
+}
+
+type DripAPIProduction struct {
+	Url string
+}
+
+type apiSubscriber struct {
+	CustomFields map[string]interface{} `json:"custom_fields"`
+	Email        string                 `json:"email"`
+	UserId       string                 `json:"user_id"`
+}
+
+type apiEvent struct {
+	Action     string                 `json:"action"`
+	Email      string                 `json:"email"`
+	OccurredAt time.Time              `json:"occurred_at"`
+	Properties map[string]interface{} `json:"properties"`
 }
 
 // Identify forwards and identify call to Drip
-func (Drip) Identify(identification integrations.Identification) (err error) {
-	logrus.Errorf("NOT IMPLEMENTED: will send %#v to Drip\n", identification)
+func (d Drip) Identify(identification integrations.Identification) (err error) {
+	s := apiSubscriber{}
+	// Drip needs an email to identify the user
+	if identification.UserTraits["email"] == nil {
+		logrus.Error("Drip: Required field email is not present")
+		return errors.New("Email is required for doing a drip request")
+	} else {
+		s.Email = identification.UserTraits["email"].(string)
+	}
+
+	s.UserId = string(identification.UserID)
+
+	// Add custom attributes
+	s.CustomFields = identification.UserTraits
+	payload, err := json.Marshal(map[string][]apiSubscriber{"subscribers": []apiSubscriber{s}})
+	err = d.api.Request("POST", "subscribers", payload)
 	return
 }
 
 // Track forwards the event to Drip
-func (Drip) Track(event integrations.Event) (err error) {
-	logrus.Errorf("NOT IMPLEMENTED: will send %#v to Drip\n", event)
+func (d Drip) Track(event integrations.Event) (err error) {
+	if event.Properties["email"] == nil {
+		logrus.WithField("err", err).Error("Drip: Required field email is not present")
+		return errors.New("Email is required for doing a drip request")
+	}
+	e := apiEvent{}
+	e.Email = event.Properties["email"].(string)
+	event.Properties["forwardlyticsReceivedAt"] = event.ReceivedAt
+	e.Action = event.Name
+	e.OccurredAt = time.Unix(event.Timestamp, 0)
+	e.Properties = event.Properties
+	payload, err := json.Marshal(map[string][]apiEvent{"events": []apiEvent{e}})
+	if err != nil {
+		logrus.WithField("err", err).Fatal("Error marshalling drip event to json")
+	}
+	err = d.api.Request("POST", "events", payload)
 	return
 }
 
 // Enabled returns wether or not the Drip integration is enabled/configured
 func (Drip) Enabled() bool {
-	return false
+	return apiToken() != "" && accountID() != ""
+}
+
+func (api DripAPIProduction) Request(method string, endpoint string, payload []byte) (err error) {
+	apiUrl := api.Url + endpoint
+	req, err := http.NewRequest(method, apiUrl, bytes.NewBuffer(payload))
+	req.SetBasicAuth(apiToken(), "")
+	req.Header.Set("User-Agent:", "forwardlytics")
+	req.Header.Set("Content-Type", "application/vnd.api+json")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		logrus.WithField("err", err).Error("Error sending request to Drip api")
```

Comment:
we should probably return from here right? The next few lines will crash I guess if we don't `return`?